pkg/diff/policy_sql_generator.go

```diff
@@ -0,0 +1,319 @@
+	}
+	return []Statement{{
+		DDL:         fmt.Sprintf("DROP POLICY %s ON %s", p.EscapedName, psg.table.GetFQEscapedName()),
+		Timeout:     statementTimeoutDefault,
+		LockTimeout: lockTimeoutDefault,
+		Hazards:     []MigrationHazard{hazard},
+	}}, nil
+}
+
+func (psg *policySQLVertexGenerator) Alter(diff policyDiff) ([]Statement, error) {
+	oldCopy := diff.old
+
+	// alterPolicyParts represents the set of strings to include in the ALTER POLICY ... ON TABLE ... statement
+	var alterPolicyParts []string
+
+	if !cmp.Equal(oldCopy.AppliesTo, diff.new.AppliesTo) {
+		alterPolicyParts = append(alterPolicyParts, fmt.Sprintf("TO %s", strings.Join(diff.new.AppliesTo, ", ")))
+		oldCopy.AppliesTo = diff.new.AppliesTo
+	}
+
+	if oldCopy.UsingExpression != diff.new.UsingExpression && diff.new.UsingExpression != "" {
+		// Weirdly, you can't actually drop a "USING EXPRESSION" clause from an ALL policy even though you
+		// can have an ALL policy with only a check expression.
+		alterPolicyParts = append(alterPolicyParts, fmt.Sprintf("USING (%s)", diff.new.UsingExpression))
+		oldCopy.UsingExpression = diff.new.UsingExpression
+	}
+
+	if oldCopy.CheckExpression != diff.new.CheckExpression && diff.new.CheckExpression != "" {
+		// Same quirk as above with ALL policies.
+		alterPolicyParts = append(alterPolicyParts, fmt.Sprintf("WITH CHECK (%s)", diff.new.CheckExpression))
+		oldCopy.CheckExpression = diff.new.CheckExpression
+	}
+	oldCopy.Columns = diff.new.Columns
+
+	if diff := cmp.Diff(oldCopy, diff.new); diff != "" {
+		return nil, fmt.Errorf("unsupported diff %s: %w", diff, ErrNotImplemented)
+	}
+
+	if len(alterPolicyParts) == 0 {
+		// There is no diff
+		return nil, nil
+	}
+
+	sb := strings.Builder{}
+	sb.WriteString(fmt.Sprintf("ALTER POLICY %s ON %s\n\t", diff.new.EscapedName, psg.table.GetFQEscapedName()))
+	sb.WriteString(strings.Join(alterPolicyParts, "\n\t"))
+
+	return []Statement{{
+		DDL:         sb.String(),
+		Timeout:     statementTimeoutDefault,
+		LockTimeout: lockTimeoutDefault,
+		Hazards:     []MigrationHazard{migrationHazardPolicyAltered},
+	}}, nil
+}
+
+func (psg *policySQLVertexGenerator) GetSQLVertexId(p schema.Policy) string {
+	return buildPolicyVertexId(psg.table.SchemaQualifiedName, p.EscapedName)
+}
+
+func buildPolicyVertexId(owningTable schema.SchemaQualifiedName, policyEscapedName string) string {
+	return buildVertexId("policy", fmt.Sprintf("%s.%s", owningTable.GetFQEscapedName(), policyEscapedName))
+}
+
+func (psg *policySQLVertexGenerator) GetAddAlterDependencies(newPolicy, oldPolicy schema.Policy) ([]dependency, error) {
+	deps := []dependency{
+		mustRun(psg.GetSQLVertexId(newPolicy), diffTypeDelete).before(psg.GetSQLVertexId(newPolicy), diffTypeAddAlter),
+	}
+
+	newTargetColumns, err := getTargetColumns(newPolicy.Columns, psg.newSchemaColumnsByName)
+	if err != nil {
+		return nil, fmt.Errorf("getting target columns: %w", err)
+	}
+
+	// Run after the new columns are added/altered
```

Comment:
> Policies on partitions must be dropped before the base table is altered, otherwise  the SQL could fail because, e.g., the policy references a column that no longer exists.

Is this quirk from above covered by the column dependencies?

Reply:
Ah maybe it's handled by how the policy generator is called by the table generator?

Reply:
Yes, this is a quirk covered by the column dependencies. It's also the same reason we can't easily support partition-level RLS policies yet